frame/collective/src/lib.rs

```diff
@@ -705,42 +721,42 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 			Error::<T, I>::WrongProposalWeight
 		);
 
-		let proposal_hash = T::Hashing::hash_of(&proposal);
-		ensure!(!<ProposalOf<T, I>>::contains_key(proposal_hash), Error::<T, I>::DuplicateProposal);
-
-		let active_proposals =
-			<Proposals<T, I>>::try_mutate(|proposals| -> Result<usize, DispatchError> {
-				proposals.try_push(proposal_hash).map_err(|_| Error::<T, I>::TooManyProposals)?;
-				Ok(proposals.len())
-			})?;
+		let proposal_bounded = T::Preimages::bound(*proposal)?;
+		ensure!(!<Voting<T, I>>::contains_key(&proposal_bounded), Error::<T, I>::DuplicateProposal);
+		ensure!(<Voting<T, I>>::count() < T::MaxProposals::get(), Error::<T, I>::TooManyProposals);
 
 		let index = Self::proposal_count();
 		<ProposalCount<T, I>>::mutate(|i| *i += 1);
-		<ProposalOf<T, I>>::insert(proposal_hash, proposal);
 		let votes = {
 			let end = frame_system::Pallet::<T>::block_number() + T::MotionDuration::get();
-			Votes { index, threshold, ayes: vec![], nays: vec![], end }
+			Votes {
+				index,
+				threshold,
+				ayes: vec![].try_into().expect("empty vec is always bounded; qed"),
+				nays: vec![].try_into().expect("empty vec is always bounded; qed"),
```

Comment:
```suggestion
				ayes: Default::default(),
				nays: Default::default(),
```
Does this not work? `Expect` and `unwrap` should be avoided as much as possible.